### PR TITLE
Port frameReceiver code from Hexitec

### PR DESCRIPTION
**Pull request by dw96**

### CODE REVIEW

**Comment by dw96** [comment]
I am of two minds whether this is good enough or whether I could shrink the code a bit further. 
Specifically, is there any relationship between the port(s) specified from the command line/config file (set in `FrameReceiverApp`) and `MercuryFrameDecoder`'s `CONFIG_FEM_PORT_MAP` ?
If `CONFIG_FEM_PORT_MAP` isn't needed then that has implications for streamlining `current_packet_fem_map_` and `fem_port_map_`.

File: data/frameReceiver/src/MercuryFrameDecoder - Copy.cpp [hunk 1]
```diff
@@ -0,0 +1,775 @@
+/*
```
Comment (ANeaves) [inline data/frameReceiver/src/MercuryFrameDecoder - Copy.cpp]: Whats this copied file for?

Comment (dw96) [reply data/frameReceiver/src/MercuryFrameDecoder - Copy.cpp]: Yeah you are right, that should never have been added in the first place. It has been removed now.

**Review by ANeaves — APPROVED** [review]

**Comment by timcnicholls** [comment]
> I am of two minds whether this is good enough or whether I could shrink the code a bit further.
> Specifically, is there any relationship between the port(s) specified from the command line/config file (set in `FrameReceiverApp`) and `MercuryFrameDecoder`'s `CONFIG_FEM_PORT_MAP` ?
> If `CONFIG_FEM_PORT_MAP` isn't needed then that has implications for streamlining `current_packet_fem_map_` and `fem_port_map_`.

Yes, the configuration FEM-PORT mapping is propagated up to the plugin. This is the mechanism for allowing the decoder plugin to know which data stream is associated with each FEM (and hence which geographical region of the image) without inspecting the source address information in the incoming packet header.

**Review by timcnicholls — APPROVED** [review]
Once you're happy with the file tidy-up this seems OK to me 👍

**Pull request merged**